tardis/energy_input/calculate_opacity.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import astropy.units as u
+
+from tardis import constants as const
+from tardis.energy_input.util import kappa_calculation
+
+MASS_SI = 28.085 * const.m_p.to(u.g).value
+MASS_FE = 55.845 * const.m_p.to(u.g).value
+M_P = const.m_p.to(u.g).value
+SIGMA_T = const.sigma_T.cgs.value
+
+
+def compton_opacity_calculation(energy, ejecta_density):
+    """Calculate the Compton scattering opacity for a given energy
+
+    Parameters
+    ----------
+    energy : float
+        The energy of the packet
+    ejecta_density : float
+        The density of the ejecta
+
+    Returns
+    -------
+    float
+        The Compton scattering opacity
+    """
+    kappa = kappa_calculation(energy)
+
+    a = 1.0 + 2.0 * kappa
+
+    sigma_KN = (
+        3.0
+        / 4.0
+        * SIGMA_T
+        * (
+            (1.0 + kappa)
+            / kappa ** 3.0
+            * ((2.0 * kappa * (1.0 + kappa)) / a - np.log(a))
+            + 1.0 / (2.0 * kappa) * np.log(a)
+            - (1.0 + 3 * kappa) / a ** 2.0
+        )
+    )
+
+    return ejecta_density / (M_P * 2) * sigma_KN
+
+
+def photoabsorption_opacity_calculation(
+    energy, ejecta_density, iron_group_fraction
+):
+    """Calculates photoabsorption opacity for a given energy
+
+    Parameters
+    ----------
+    energy : float
+        Packet energy
+    ejecta_density : float
+        The density of the ejecta
+    iron_group_fraction : float
+        Fraction of iron group elements in the shell
+
+    Returns
+    -------
+    float
+        Photoabsorption opacity
```

Comment:
I think it would be good to have a reference here. We should probably also mention that this is an approximate treatment.